deadlines-api/src/main/java/com/palantir/deadlines/api/RemainingDeadline.java

```diff
@@ -16,4 +16,6 @@
 
 package com.palantir.deadlines.api;
 
-public class DeadlineExpiredException extends RuntimeException {}
+import java.time.Duration;
+
+public record RemainingDeadline(Duration value, boolean internal) {}
```

Comment:
Is the `internal` flag relevant to callers? If we encapsulate the exception creation on deadline expiry, we can expose that detail through the exception if a failure occurs, and we don't need to expose this state.

Reply:
i guess that's a good point - you're saying we will (eventually) need to e.g. `throw InternalDeadlineExpired` or `throw ExternalDeadlineExpired` for instance?

Reply:
fixed

Reply:
👍